### ISSUE TITLE
There's no "safe" way to use Freetype

### ISSUE THREAD

**Issue by ruby3141** [imgui-rs/imgui-rs#773]
This issue contains long trivia about how I found the reason why it didn't work.
I was dead confused while solving this issue, so I elongated the explanation hoping to save some tortured souls.

If you just want to know how to enable Freetype right now, go to the end of problem description.

### Is your feature request related to a problem? Please describe.

I tried using a new font (NanumSquare Neo, for those who wants to know), but it looks blurish and ugly.
So I enabled Freetype feature of `imgui` (and `imgui-sys` manually to use vcpkg),
```toml
imgui-sys = { version = "0.11.0", features = ["freetype", "use-vcpkg"] }
imgui = { version = "0.11.0", features = ["freetype"] }
```
but found no difference. The two result was pixel perfect identical.

Found this line on Imgui Freetype readme, so I confired that Freetype isn't activated, by changing oversampling settings.

https://github.com/imgui-rs/imgui-rs/blob/cd97e171fe046c063a6e731d6819a887e48d985d/imgui-sys/third-party/imgui-master-freetype/imgui/misc/freetype/README.md?plain=1#L23-L25

After many many tries of changing, rebuilding, trying, I digged into imgui-sys, insert some test codes on cpp files,
and finally found out `IMGUI_ENABLE_FREETYPE` is not defined when imgui main parts are compiling.

![Screenshot 2024-04-22 114307](https://github.com/imgui-rs/imgui-rs/assets/3256382/a42d86da-20c0-471f-8848-b049426a4aba)

`build.rs` said `IMGUI_ENABLE_FREETYPE` sets with `CIMGUI_FREETYPE`, and it was.

https://github.com/imgui-rs/imgui-rs/blob/cd97e171fe046c063a6e731d6819a887e48d985d/imgui-sys/build.rs#L84-L86

https://github.com/imgui-rs/imgui-rs/blob/cd97e171fe046c063a6e731d6819a887e48d985d/imgui-sys/third-party/imgui-master-freetype/cimgui.cpp#L4-L17

But `cimgui.cpp` compiles after Imgui things. It has no effect while building Imgui main parts.

https://github.com/imgui-rs/imgui-rs/blob/cd97e171fe046c063a6e731d6819a887e48d985d/imgui-sys/include_imgui_master_freetype.cpp#L1-L12

If `IMGUI_ENABLE_FREETYPE` is not set during compilation of Imgui, `imstb` is included and used as default.

https://github.com/imgui-rs/imgui-rs/blob/cd97e171fe046c063a6e731d6819a887e48d985d/imgui-sys/third-party/imgui-master-freetype/imgui/imgui_internal.h#L106-L108

https://github.com/imgui-rs/imgui-rs/blob/cd97e171fe046c063a6e731d6819a887e48d985d/imgui-sys/third-party/imgui-master-freetype/imgui/imgui_draw.cpp#L2268-L2271

Since Freetype builder is built and included but not set as default,
it needs to be exchanged at runtime, before building and loading font atlas textures.
And I think, with current release of `imgui-sys`, that there's no other way than `unsafe`ly doing it.
```rs
unsafe {
    ctx.fonts().raw_mut().FontBuilderIO = ImGuiFreeType_GetBuilderForFreeType();
}
```

**TL;DR**
1. Imgui-rs doesn't set Freetype atlas builder as default even if the feature is enabled.
   (And I think it's misleading.)
2. For now, you can use Freetype by changing FontBuilderIO at runtime.
3. But you should do it `unsafe`ly.

### Describe the solution you'd like
1. Make Freetype as default when the feature is enabled.

### Describe alternatives you've considered
1. Make way to exchange font builder without `unsafe`ly using `imgui-sys`

### Additional context
Maybe having example program about using Freetype would be very helpful IMO.

**Comment by sanbox-irl** [comment]
Wow, this is really an *excellent* overview! I think your solution (ie, we should just change the default loaded builder_io) is correct.
